### PR TITLE
Remove unconfigurable_constants module

### LINKED ISSUE #16107: Force "unconfigurable constants" to top level
There are still several places in the code base where we are using the library `node_config_unconfigurable_constants`. We should stop accessing this directly, instead going through on of the sanctioned config records

- `constraint_constants`
- `genesis_constants`
- `compile_config`

### PR DESCRIPTION
**Pull request by martyall**
Depends on #16186 

Fixes #16107 

## The problem
We have this `mina_node_config.unconfigurable_constants` package that is leaking out the usage of `mina_node_config`. We need to access these fields via the standard `*_constants` records.

## Solution
- Delete `mina_node_config.unconfigurable_constants`
- Remove all direct access to `Node_config` values in the codebase outside of `signature_kind`, `genesis_constants` and `mina_compile_config` modules 
- Many of the `unconfigurable_constants` were duplicated in both `Mina_compile_config` and `Genesis_constants`. It seems more natural to keep them in one place and `Genesis_constants` is the obvious choice. Remove the duplicated fields.

### CODE REVIEW

**Comment by martyall** [comment]
!ci-nightly-me

**Comment by martyall** [comment]
nightly results https://buildkite.com/o-1-labs-2/mina-end-to-end-nightlies/builds/2683

File: src/lib/node_config/for_unit_tests/node_config_for_unit_tests.ml [hunk 1]
```diff
@@ -64,3 +63,40 @@ let (vrf_poll_interval : int) = (0 : int)
...
+
+let zkapp_transaction_cost_limit = 69.45
+
+let max_event_elements = 100
+
+let max_action_elements = 100
+
+let zkapp_cmd_limit_hardcap = 128
+
+(* These are fine to be non-configurable *)
```
Comment (georgeee) [inline src/lib/node_config/for_unit_tests/node_config_for_unit_tests.ml:94]:
```suggestion
```

File: src/app/cli/src/init/client.ml [hunk 2]
```diff
@@ -314,10 +358,19 @@ let verify_receipt =
...
   in
+  let config_file = Cli_lib.Flag.conf_file in
   Command.async ~summary:"Verify a receipt of a sent payment"
     (Cli_lib.Background_daemon.rpc_init
-       (Args.zip4 payment_path_flag proof_path_flag address_flag token_flag)
-       ~f:(fun port (payment_path, proof_path, pk, token_id) ->
+       (Args.zip5 config_file payment_path_flag proof_path_flag address_flag
+          token_flag )
+       ~f:(fun port (config_file, payment_path, proof_path, pk, token_id) ->
+         let%bind compile_config =
```
Comment (georgeee) [inline src/app/cli/src/init/client.ml:367]: Code below is repeated more than ten times, probably we could remove the duplication.

**Review by georgeee — APPROVED** [review]

**Review by georgeee — APPROVED** [review]

**Comment by georgeee** [comment]
!ci-build-me

**Pull request merged**